core/domain/state_domain.py

```diff
@@ -1925,6 +1928,341 @@ def validate(self) -> None:
+                'attribute.'
+            )
+
+        if (
+            tag['start-with-value'].strip() in
+            ('&quot;&quot;', '', '\'\'', '\"\"')
+        ):
+            raise utils.ValidationError(
+                'Video tag \'start-with-value\' attribute should not'
+                'be empty.'
+            )
+
+        if not tag.has_attr('end-with-value'):
+            raise utils.ValidationError(
+                'Video tag does not have \'end-with-value\' '
+                'attribute.'
+            )
+
+        if (
+            tag['end-with-value'].strip() in
+            ('&quot;&quot;', '', '\'\'', '\"\"')
+        ):
+            raise utils.ValidationError(
+                'Video tag \'end-with-value\' attribute should not'
+                'be empty.'
+            )
+
+        start_value = tag['start-with-value'].strip()
+        end_value = tag['end-with-value'].strip()
+        if start_value > end_value and start_value != '0' and end_value != '0':
+            raise utils.ValidationError(
+                'Start value should not be greater than End value in Video tag.'
+            )
+
+        if not tag.has_attr('autoplay-with-value'):
+            raise utils.ValidationError(
+                'Video tag does not have \'autoplay-with-value\' '
+                'attribute.'
+            )
+
+        if tag['autoplay-with-value'].strip() not in (
+            'true', 'false', '\'true\'', '\'false\'',
+            '\"true\"', '\"false\"'
+        ):
+            raise utils.ValidationError(
+                'Video tag \'autoplay-with-value\' attribute should be '
+                'a boolean value.'
+            )
+
+        if not tag.has_attr('video_id-with-value'):
+            raise utils.ValidationError(
+                'Video tag does not have \'video_id-with-value\' '
+                'attribute.'
+            )
+
+        if tag['video_id-with-value'] is None:
+            raise utils.ValidationError(
+                'Video tag \'video_id-with-value\' attribute should not'
+                'be None.'
+            )
+        elif (
+            tag['video_id-with-value'].strip() in
+            ('&quot;&quot;', '', '\'\'', '\"\"')
+        ):
+            raise utils.ValidationError(
+                'Video tag \'video_id-with-value\' attribute should not'
+                'be empty.'
+            )
+
+    for tag in soup.find_all('oppia-noninteractive-link'):
+        if not tag.has_attr('text-with-value'):
+            raise utils.ValidationError(
+                'Link tag does not have \'text-with-value\' '
+                'attribute.'
+            )
+
+        empty_values = ['&quot;&quot;', '', '\'\'', '\"\"']
+        if tag['text-with-value'].strip() in empty_values:
+            raise utils.ValidationError(
+                'Link tag \'text-with-value\' attribute should not'
+                'be empty.'
+            )
+
+        if not tag.has_attr('url-with-value'):
```

Comment:
Wasn't there also the 'cannot be http link' validation? What list of validations are you referencing for this PR? I have a feeling you aren't looking at the right list.

Reply:
This one check will be done by @Lawful2002, you can take a look at his PR [here](https://github.com/oppia/oppia/pull/15955). I'm following [this](https://docs.google.com/spreadsheets/d/1kbjCeADo0VuP7ezSt84NOguA-z9tPJ2IokGGZuh1-Ts/edit?usp=sharing) list that I created for M1, I think now all checks are covered.